From 4b45a2a9b2082bc1aac12732b3e2c5ea68e83b0d Mon Sep 17 00:00:00 2001
From: Przemyslaw Czarnota <pczarnota@maxlinear.com>
Date: Thu, 15 Jul 2021 15:38:30 +0200
Subject: [PATCH] PONRTSYS-9098: Add support for mapping extracted packets to
 CPU queue

- Use subif instead of subif_groupid

Use of subif_groupid for queue mapping is incorrect. It works only by
accident, because pon_qos_get_subif_idx() is called now only for
indev and indev is currently only a GEM port. For the PON side subif
equals subif_groupid, and this is why it works.

This change changes pon_qos_get_subif_idx() to use subif instead
of subif_groupid.
---
 drivers/net/datapath/pon_qos/pon_qos_tc_qmap.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/datapath/pon_qos/pon_qos_tc_qmap.c b/drivers/net/datapath/pon_qos/pon_qos_tc_qmap.c
index 979e4bf53409..bf843d726382 100644
--- a/drivers/net/datapath/pon_qos/pon_qos_tc_qmap.c
+++ b/drivers/net/datapath/pon_qos/pon_qos_tc_qmap.c
@@ -156,7 +156,7 @@ static int pon_qos_get_subif_idx(struct net_device *dev, int *subif)
 		return -ENODEV;
 	}
 
-	*subif = dp_subif.subif_groupid;
+	*subif = dp_subif.subif;
 
 	return 0;
 }
